enterprise/views.py

```diff
@@ -226,299 +242,133 @@ def get_default_context(self, enterprise_customer, request):
-        """
-        if not CourseApiClient().get_course_details(course_id):
-            raise Http404
-
-        enrollment_deferred = request.POST.get('enrollment_deferred')
-        if enrollment_deferred is None:
-            enterprise_customer = get_enterprise_customer_for_user(request.user)
-            enterprise_customer_user, __ = EnterpriseCustomerUser.objects.get_or_create(
-                enterprise_customer=enterprise_customer,
-                user_id=request.user.id
-            )
-            EnterpriseCourseEnrollment.objects.update_or_create(
-                enterprise_customer_user=enterprise_customer_user,
-                course_id=course_id,
-            )
-            DataSharingConsent.objects.update_or_create(
-                username=request.user.username,
-                course_id=course_id,
-                enterprise_customer=enterprise_customer,
-                defaults={
-                    'granted': consent_provided
-                },
-            )
-
-        if not consent_provided:
-            failure_url = request.POST.get('failure_url') or reverse('dashboard')
-            return redirect(failure_url)
-
-        return redirect(request.POST.get('redirect_url', reverse('dashboard')))
-
     @method_decorator(login_required)
-    def post_program_specific_consent(self, request, program_uuid, consent_provided):
+    def post(self, request):
         """
-        Interpret the program-specific form above and save it to an EnterpriseCourseEnrollment object.
+        Process the above form.
         """
-        if not CourseCatalogApiServiceClient.program_exists(program_uuid):
-            raise Http404
-
         enterprise_uuid = request.POST.get('enterprise_customer_uuid')
-        failure_url = request.POST.get('failure_url')
         success_url = request.POST.get('redirect_url')
-        enrollment_deferred = request.POST.get('enrollment_deferred')
-        username = request.user.username
-
-        if not (enterprise_uuid and failure_url and success_url):
+        failure_url = request.POST.get('failure_url')
+        if not (enterprise_uuid and success_url and failure_url):
             raise Http404
 
-        consent_record = get_data_sharing_consent(username, enterprise_uuid, program_uuid=program_uuid)
+        course_id = request.POST.get('course_id', '')
+        program_uuid = request.POST.get('program_uuid', '')
+        if not self.course_or_program_exist(course_id, program_uuid):
+            raise Http404
 
+        consent_record = get_data_sharing_consent(
+            request.user.username,
+            enterprise_uuid,
+            program_uuid=program_uuid,
+            course_id=course_id
+        )
         if consent_record is None:
             raise Http404
 
-        if enrollment_deferred is None and consent_record.consent_required():
+        enterprise_customer = get_enterprise_customer_for_user(request.user)
```

Comment:
Can we not get the enterprise customer off the consent record?

Reply:
Oh yeah, we can :+1: